test/test_helper.rb

```diff
@@ -71,6 +79,36 @@ def Minitest.after_run(&block)
 require 'fixtures/poro'
 
 ActiveSupport.on_load(:active_model_serializers) do
-  $action_controller_logger = ActiveModelSerializers.logger # rubocop:disable Style/GlobalVars
+  $action_controller_logger = ActiveModelSerializers.logger
   ActiveModelSerializers.logger = Logger.new(IO::NULL)
 end
+
+# From:
+# https://github.com/seattlerb/minitest/blob/644a52fd0/lib/minitest/unit.rb#L768-L787
+# https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest.rb#L45-L59
+# But we've replaced `at_exit` with `END` called before the 'at_exit' hook.
+class MiniTestHack
+  def self.autorun(minitest_run)
+    # don't run if there was a non-exit exception
+    return if $! and not ($!.kind_of? SystemExit and $!.success?)
+
+    # the order here is important. The at_exit handler must be
+    # installed before anyone else gets a chance to install their
+    # own, that way we can be assured that our exit will be last
+    # to run (at_exit stacks).
+    exit_code = nil
+
+    trap('EXIT') do
```

Comment:
We can't use `END` inside a method, but trapping EXIT is just fine, especially since we don't need to run with any other signal